pkg/reconciler/instances/istio/merge/merge.go

```diff
@@ -3,23 +3,54 @@ package merge
 import (
 	"context"
 	"encoding/json"
+	"strconv"
 
 	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/istio/clientset"
 	"github.com/kyma-project/istio/operator/api/v1alpha1"
 	"github.com/kyma-project/istio/operator/pkg/lib/gatherer"
 	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/wrapperspb"
 	istioOperator "istio.io/istio/operator/pkg/apis/istio/v1alpha1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	apiMeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-// IstioOperatorConfiguration merges default Kyma Istio Operator file with user configuration in Istio CR.
-// If there is no IstioCRD or there are no Istio CR present on the cluster, it defaults to the operator file.
+const (
+	KymaNamespace = "kyma-system"
+	ConfigMapCNI  = "kyma-istio-cni"
+)
+
+// IstioOperatorConfiguration merges default Kyma Istio Operator file with user configuration in Istio CR and in Istio ConfigMap.
+// If there is no IstioCRD or there are no Istio CR/ConfigMap present on the cluster, it defaults to the operator file.
 func IstioOperatorConfiguration(ctx context.Context, provider clientset.Provider, operatorManifest string, kubeConfig string, logger *zap.SugaredLogger) (string, error) {
 	istioCRList, err := getIstioCR(ctx, provider, kubeConfig)
 	if err != nil {
 		return "", err
 	}
 
+	kubeClient, err := provider.RetrieveFrom(kubeConfig, logger)
+	if err != nil {
+		return "", err
+	}
+
+	cniEnabled, err := GetCNIConfigMap(ctx, kubeClient)
+	if err != nil {
+		return "", err
+	}
+
+	if cniEnabled != "" {
+		combinedManifest, err := applyIstioCNI(cniEnabled, operatorManifest)
+		if err != nil {
+			logger.Error("Could not apply Istio CNI ConfigMap into Istio Operator")
+			return "", err
+		}
+
+		logger.Debugf("Istio CNI ConfigMap was applied to the Istio Operator configuration")
+		return combinedManifest, nil
```

Comment:
If I understand it correctly this return will cause, that IstioCR is not applied when CNI is enabled. I think the CR must be applied, even if CNI is enabled.